minion-certificate-verifier/main/src/main/java/org/opennms/horizon/minioncertverifier/controller/CertificateController.java

```diff
@@ -68,12 +76,30 @@ public ResponseEntity<Void> validateDebug(HttpServletRequest request) throws Exc
             logger.info("\t{}: {}", header, request.getHeader(header));
         }
 
-        String subjectDn = request.getHeader(headerName);
-        return validate(subjectDn);
+        String cert = request.getHeader("ssl-client-cert");
+        return validate(cert);
     }
 
     @GetMapping
-    public ResponseEntity<Void> validate(@RequestHeader("ssl-client-subject-dn") String clientSubjectDn) throws Exception {
+    public ResponseEntity<Void> validate(@RequestHeader("ssl-client-cert") String certificatePem) {
+        var span = Span.current();
+        CertificateParser parser;
+        try {
+            parser = new CertificateParser(certificatePem);
+        } catch (CertificateException ex) {
+            span.setAttribute("message", "Invalid certificate");
```

Comment:
What you want to do here is record the exception using [`recordException`](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/exceptions/#recording-an-exception). Note: we don't need to mark this as an "escaped exception" because we don't re-throw the exception.
--


<br class="Apple-interchange-newline">

```suggestion
            span.recordException(ex);
            span.setStatus(StatusCode.ERROR, "invalid certificate: " + ex);
```

Reply:
If it might be useful, you could also add the problematic certificate itself as an attribute. Actually, the more I think about it, I think it would be a good idea. Then, if you need to debug a parsing issue, you have *everything* you need here to do so.

Reply:
done